lib/tasks/data_key_validation.rake

```diff
@@ -0,0 +1,52 @@
+
+namespace :data_key_validation do
+  desc "Validates the value of ENV['CONJUR_DATA_KEY']"
+  # This task validate the master data key which is stored in `ENV['CONJUR_DATA_KEY']`
+  # Once the db is created, a secret will be created using the master key.
+  # Each time the server is restarted this task will attempt to retrieve the
+  # secret, if the provided key does not match the original master the secret retrieval
+  # will fail.
+  #
+  # What's the 'environment' task in Rake?
+  # the :validate => :environment means "do environment before validate".
+  # You can get access to your models, and in fact, your whole environment
+  # by making tasks dependent on the environment task.
+  task :validate => [:environment] do
+    data_key = ENV['CONJUR_DATA_KEY']
+
+    if data_key.nil? || data_key.empty?
+      raise Errors::System::EmptyConjurDataKey
+      exit 1
+    end
+
+    # try get the master key validator resource
+    resource_id = "!:!:data-key-validator-resource"
+
+    unless Resource[resource_id]
+      # create the master key validator resource
+      # and store a secret using the master key
+      role_id = "!:!:data-key-validator"
+
+      begin
+        Role.create(role_id: role_id)
+        Resource.create(resource_id: resource_id, owner_id: role_id)
+        Secret.create(resource_id: resource_id, value: "Don't Panic.")
+      rescue => e
+        $stderr.puts "Error creating a resource for rake: data_key_validation, Error: #{e.inspect}"
+      end
+    else
+      begin
+        # Attempt to retrieve the secret encrypt with the original data key.
+        # A failure to decrypt the value will indicate the key is invalid.
+        Resource[resource_id].secret.value
+      rescue => e
+        if e.inspect == 'OpenSSL::Cipher::CipherError'
+          raise Errors::System::InvalidConjurDataKey
+        else
+          $stderr.puts "rake: data_key_validation, error retrieving Resource[#{resource_id}].secret.value, #{e.inspect}"
```

Comment:
What is the value in printing the secret value? How does it help the user troubleshoot the problem?

Reply:
It does not print the secret value.

Reply:
i would change it to have words instead of code. `error retrieving data-key validator secret`

Reply:
also - why is this not in `errors.rb`?

Reply:
> also - why is this not in `errors.rb`?

I don't expect this error to rise, it's a fallback.